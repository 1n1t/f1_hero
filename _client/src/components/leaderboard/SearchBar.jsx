import React from 'react';

const SearchBar = props => {
  return (
    <div className="search-bar">
      <input
        type="text"
        placeholder="Search..."
        onChange={props.onChange}
        value={props.searchTerm}
      />
      <i className="fa fa-search" aria-hidden="true"></i>
    </div>
  );
};

export default SearchBar;
