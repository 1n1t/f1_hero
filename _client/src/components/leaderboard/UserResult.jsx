import React, {Component} from 'react';

class UserResult extends Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    const {index, user, scores} = this.props.score.score;

    return (
      <div className="user-result">
        <div>{index}.</div>
        <div className="username">{user}</div>
        <div>{scores}pt</div>
      </div>
    );
  }
}

export default UserResult;