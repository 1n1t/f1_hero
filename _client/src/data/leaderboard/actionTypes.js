/* eslint-disable import/prefer-default-export */
export const UPDATE_SCORES = 'leaderboard/UPDATE_SCORES';
