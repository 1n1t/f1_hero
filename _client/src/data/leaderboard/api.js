/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getUsersScores: '/api/leaderboard'
};

export const getUsersScores = () => fetchApi(endPoints.getUsersScores);
