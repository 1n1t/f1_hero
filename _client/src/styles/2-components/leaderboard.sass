.leaderboard
  width: 400px
  background-color: $color-black
  color: white
  margin: 0 auto
  padding: 20px

.user-result
  display: flex
  padding: 0 30px
  &:not(:last-child)
    margin-bottom: 20px

  .username
    flex-grow: 1
    margin-left: 20px