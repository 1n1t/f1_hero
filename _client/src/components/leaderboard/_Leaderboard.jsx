import React, { Component } from 'react';
import axios from 'axios';

import UserResult from './UserResult';

class Leaderboard extends Component {
  async componentDidMount() {
    const scores = (await axios.get('/api/leaderboard')).data;
    this.setState({ scores });
  }

  render() {
    const content =
      this.state && this.state.scores
        ? this.state.scores.map((score, index) => (
            <UserResult score={{ score, index }} key={score.user} />
          ))
        : '';

    return (
      <div className="leaderboard">
        <div className="results">{content}</div>
      </div>
    );
  }
}

export default Leaderboard;
