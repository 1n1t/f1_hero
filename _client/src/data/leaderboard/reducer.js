/* eslint-disable import/prefer-default-export */
import * as actionTypes from './actionTypes';

const initialState = {
	scores: []
};

export const reducer = (state = initialState, action) => {
	switch (action.type) {
		case actionTypes.UPDATE_SCORES:
			return {
				...state,
				scores: action.payload
			};
		default:
			return state;
	}
};
