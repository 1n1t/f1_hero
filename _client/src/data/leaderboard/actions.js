/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const getScores = scores => ({
	type: actionTypes.GET_SCORES,
	payload: scores
});

export const getUsersScores = () => async dispatch => {
	const response = await api.getUsersScores();
	dispatch(getScores(response.data.leaderboard));
};
