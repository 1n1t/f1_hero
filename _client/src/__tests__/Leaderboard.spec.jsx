import React from 'react';
import {shallow} from 'enzyme';
import {Unwrapped as Leaderboard} from '../components/leaderboard/_Leaderboard';
import toJson from 'enzyme-to-json';

test('Leaderboard renders correctly', async () => {
  const component = shallow(<Leaderboard/>);
  expect(toJson(component)).toMatchSnapshot();
});
