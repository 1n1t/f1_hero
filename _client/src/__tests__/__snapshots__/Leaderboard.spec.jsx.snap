// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard renders correctly 1`] = `
<div
  className="leaderboard"
>
  <SearchBar
    onChange={[Function]}
    searchTerm=""
  />
  <div
    className="results"
  />
</div>
`;
